src/api/client.ts

```diff
@@ -178,17 +103,49 @@ export class WsClient extends Emittery {
    * @param request - The authentication request, containing an app authentication token.
    */
   async authenticate(request: AppAuthenticationRequest): Promise<void> {
-    return this.exchange(request, (request, resolve) => {
+    this.authenticationToken = request.token;
+    return this.exchange(request, (request, resolve, reject) => {
+      const invalidTokenCloseListener = (
+        closeEvent: IsoWebSocket.CloseEvent
+      ) => {
+        reject(
```

Comment:
If we're rejecting because of an invalid token, should `this.authenticationToken` not be cleared here?

Reply:
Yes, absolutely!